pages/checkOut.tsx

```diff
@@ -10,23 +10,64 @@ import {
   useMediaQuery,
   useTheme,
 } from '@mui/material'
-import { CategoryResponse } from 'utils/types'
+import {
+  CategoryResponse,
+  CheckInOutFormData,
+  InventoryItemRequest,
+  ItemDefinitionResponse,
+  UserResponse,
+} from 'utils/types'
 import { GetServerSidePropsContext } from 'next'
+import React from 'react'
+import { CheckInOutFormDataToInventoryItemRequest } from 'utils/transformations'
+import usersHandler from '@api/users'
+import itemDefinitionsHandler from '@api/itemDefinitions'
 import { apiWrapper } from 'utils/apiWrappers'
 import categoriesHandler from '@api/categories'
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
   return {
-    props: { categories: await apiWrapper(categoriesHandler, context) },
+    props: {
+      categories: await apiWrapper(categoriesHandler, context),
+      itemDefinitions: await apiWrapper(itemDefinitionsHandler, context),
+      users: await apiWrapper(usersHandler, context),
+    },
   }
 }
 interface Props {
   categories: CategoryResponse[]
+  itemDefinitions: ItemDefinitionResponse[]
+  users: UserResponse[]
 }
 
-export default function CheckOutPage({ categories }: Props) {
+export default function CheckOutPage({
+  categories,
+  itemDefinitions,
+  users,
+}: Props) {
   const theme = useTheme()
   const isMobileView = useMediaQuery(theme.breakpoints.down('sm'))
+
+  const [formData, setFormData] = React.useState<CheckInOutFormData>(
+    {} as CheckInOutFormData
+  )
+
+  const onSubmit = async (formData: CheckInOutFormData) => {
+    const inventoryItem: Partial<InventoryItemRequest> =
+      CheckInOutFormDataToInventoryItemRequest(formData)
```

Comment:
Go into this function and copy the implementation from [here](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/blob/demo-41023/utils/transformations.ts). This will fix the issues Shashank mentioned